### PR TITLE
Support NULLS FIRST in KeyCondition

### PR DESCRIPTION
**Pull request by DimasKovas**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

Detailed description / Documentation draft:
Support using index for `isNull`/`isNotNull` expressions even if the index is built with NULLS FIRST instead of the default NULLS LAST. Actually it's not possible to change NULLS FIRST/LAST in index for ClickHouse, but it is needed for some code in Yandex that is embedding ClickHouse engine.

### CODE REVIEW

**Comment by amosbird** [comment]
Description can be improved: Support `IS [NOT] NULL` operator when NULLS FIRST is specified. Also a simple test is required.

**Comment by alexey-milovidov** [comment]
@DimasKovas Tests have failed: https://clickhouse-test-reports.s3.yandex.net/29528/e3ae32f7635c82a441ef5af55cd787c2302765f0/style_check/typos_output.txt

File: src/Storages/MergeTree/KeyCondition.cpp [hunk 1]
```diff
@@ -1988,7 +1989,10 @@ BoolMask KeyCondition::checkInHyperrectangle(
             /// No need to apply monotonic functions as nulls are kept.
             bool intersects = element.range.intersectsRange(*key_range);
             bool contains = element.range.containsRange(*key_range);
+
             rpn_stack.emplace_back(intersects, !contains);
+            if (element.function == RPNElement::FUNCTION_IS_NULL)
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/KeyCondition.cpp:1994]: Please add some clarification.

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
Almost green. I will fix a typo.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**